apps/prairielearn/src/schemas/accessControl.ts

```diff
@@ -0,0 +1,238 @@
+import { z } from 'zod';
+
+import { DateFromISOString } from '@prairielearn/zod';
+
+export { DateFromISOString };
+
+// TODO: Intuition: To prevent JSON users from creating roles that don't have intended effects, it might be nice to enforce that if an *enabled field is set to NULL
+// (i.e. inherit), we disallow the value to be set.
+const createEnabledFieldValidator = (pairs: Array<[string, string]>) => {
```

Comment:
We will not validate these using Zod. I can dig up the related discussions if you are curious. Please split these out into a `validateAccessControl` function in `course-db.ts` with corrosponding tests for some sample schemas.